tests/baselines/reference/bindingPatternCannotBeOnlyInferenceSource.errors.txt

```diff
@@ -1,18 +1,22 @@
 bindingPatternCannotBeOnlyInferenceSource.ts(2,7): error TS2571: Object is of type 'unknown'.
 bindingPatternCannotBeOnlyInferenceSource.ts(3,9): error TS2339: Property 'p1' does not exist on type 'unknown'.
+bindingPatternCannotBeOnlyInferenceSource.ts(3,16): error TS2571: Object is of type 'unknown'.
 bindingPatternCannotBeOnlyInferenceSource.ts(4,7): error TS2461: Type 'unknown' is not an array type.
 bindingPatternCannotBeOnlyInferenceSource.ts(4,7): error TS2571: Object is of type 'unknown'.
 bindingPatternCannotBeOnlyInferenceSource.ts(5,7): error TS2461: Type 'unknown' is not an array type.
+bindingPatternCannotBeOnlyInferenceSource.ts(5,18): error TS2571: Object is of type 'unknown'.
 
 
-==== bindingPatternCannotBeOnlyInferenceSource.ts (5 errors) ====
+==== bindingPatternCannotBeOnlyInferenceSource.ts (7 errors) ====
     declare function f<T>(): T;
     const {} = f();       // error (only in strictNullChecks)
           ~~
 !!! error TS2571: Object is of type 'unknown'.
     const { p1 } = f();   // error
             ~~
 !!! error TS2339: Property 'p1' does not exist on type 'unknown'.
+                   ~~~
+!!! error TS2571: Object is of type 'unknown'.

```

Comment:
This gets a little bit noisy because we essentially report the same problem twice. 

There are cases in the changed baselines that require the error to be reported on the initializer, like here:
https://github.com/microsoft/TypeScript/blob/28cdebc55f42c7afb3aad44c41b10598bdab99a0/tests/baselines/reference/destructuringArrayBindingPatternAndAssignment1ES5.errors.txt#L32-L37

So if there is desire to limit the amount of reported errors I think it would be good to report consistently them on the initializer at all times. I think handling this differently for different kind of patterns will complicate the code. But if there is a strong belief that this would provide a better DX - I can implement the required changes.